test/common/common_e2e.go

```diff
@@ -0,0 +1,265 @@
+package common
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/eks"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/runner"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	ddaPrefix string = "dda-datadog"
+	dcaPrefix string = "dda-datadog-cluster-agent"
+	ccPrefix  string = "dda-datadog-clusterchecks"
+)
+
+var (
+	defaultLocalPulumiConfigs = runner.ConfigMap{
+		"ddinfra:aws/defaultKeyPairName": auto.ConfigValue{Value: os.Getenv("AWS_KEYPAIR_NAME")}}
+	defaultCIPulumiConfigs = runner.ConfigMap{
+		"aws:skipCredentialsValidation": auto.ConfigValue{Value: "true"},
+		"aws:skipMetadataApiCheck":      auto.ConfigValue{Value: "false"},
+	}
+	dcaPodLabelSelector = fmt.Sprintf("app=%s", dcaPrefix)
+	ddaPodLabelSelector = fmt.Sprintf("app=%s", ddaPrefix)
+	ccPodLabelSelector  = fmt.Sprintf("app=%s", ccPrefix)
+)
+
+type ExpectedPods struct {
+	Prefix           string
+	ContainerName    string
+	PodLabelSelector string
+	PodCount         int
+	Msg              string
+}
+
+var ExpDcaPods = ExpectedPods{
+	Prefix:           dcaPrefix,
+	ContainerName:    "cluster-agent",
+	PodLabelSelector: dcaPodLabelSelector,
+	PodCount:         1,
+	Msg:              "There should be 1 datadog-cluster-agent pod by default.",
+}
+
+var ExpDdaPods = ExpectedPods{
+	Prefix:           ddaPrefix,
+	ContainerName:    "agent",
+	PodLabelSelector: ddaPodLabelSelector,
+	Msg:              "There should be 1 datadog-agent pod per node.",
+}
+
+var ExpCcPods = ExpectedPods{
+	Prefix:           ccPrefix,
+	ContainerName:    "agent",
+	PodLabelSelector: ccPodLabelSelector,
+	PodCount:         2,
+	Msg:              "There should be 2 datadog-cluster-check pods by default.",
+}
```

Comment:
Are these variable names easy to read/understand? Since they define the expected results, should they be placed in the same file as the tests?